pkg/sql/copy.go

```diff
@@ -387,6 +387,15 @@ func (c *copyMachine) readTextData(ctx context.Context, final bool) (brk bool, e
 }
 
 func (c *copyMachine) readCSVData(ctx context.Context, final bool) (brk bool, err error) {
+	// Don't try to parse the CSV until we read all the data. This is because
```

Comment:
should we look at improving this in future / make an issue?

Reply:
as @steven-sheehy says this behavior isn't that good, and it might even be better to leave the CSV parsing as-is instead of doing this workaround

i'll give a real fix another shot.